core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -1190,8 +1190,10 @@ class Partition(val topicPartition: TopicPartition,
       case Some(leaderLog) =>
         // keep the current immutable replica list reference
         val curMaximalIsr = partitionState.maximalIsr
-
-        if (isTraceEnabled) {
```

Comment:
It already has traceEnabled check, why they will cost CPU

Reply:
this isTraceEnabled is a method which will query the log4j or log implementation to check the log level 
<img width="1858" alt="image" src="https://github.com/user-attachments/assets/a3668cb8-e405-41ec-9501-0bec1032739e">
when high rate check tryComplete the method will burst cpu.

Reply:
this patch only disable the two part trace check with call the method to check the log implementation log level.
<img width="1854" alt="image" src="https://github.com/user-attachments/assets/4cf45269-d583-4505-ac3c-62167c4a328d">

Reply:
Could we check isTraceEnabled and cache the result instead of introduce AUTOMQ_ENABLE_TRACE_LOG switch